### ISSUE TITLE
"amount" filter has wrong semantics for transfers

### ISSUE THREAD

**Issue by Bone008** [Bone008/finance-tracker#171]
A transfer of 200 from one account to another should be matched by "amount:200".

"amount=0" should probably also match it, if the user is thinking about the concept of "billed amount".

**Comment by Taimonania** [comment]
A potential use-case to keep in mind: for e.g. money lending or withdrawing cash, I group the transaction such that they have an amount of 0. By filtering with "-amount:0" I can easily see which transactions weren't grouped with a counterpart (e.g. who still ows me money?). 

This change would break this filtering, right? Because I would then also get all transfers (grouped transactions across accounts) that have an amount of 0.

**Comment by Bone008** [comment]
@Taimonania Good point. I plan on letting `amount` still match the value of the whole transaction group, in addition to individual transactions, so `amount:0` would still work unchanged.

`amount:200` will start matching all of the following examples:

 1. a single transaction of (+|-)200€
 2. a group of -150€ and -50€
 3. a group of -200€ and +200€ (which might be a transfer)
 4. a group of (+|-)200€ and 42€

Before the change, it only matched (1) and (2). I think matching (3) and (4) is reasonable.

**Comment by Bone008** [comment]
This change introduced a new problem where (in particular in analytics) you can no longer restrict results to income/expenses only with `amount>0` / `amount<0`, because grouped transactions are included too.

I should probably restrict the new behavior to `:`, which will then hopefully (?!) always to the right thing.

No idea if `amount>150` should match a transfer of 200€ from A to B (billed at 0) though. Argh.
